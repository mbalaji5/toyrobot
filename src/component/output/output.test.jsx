import { render, screen } from "@testing-library/react";
import { Output } from ".";

describe("Should validate Error component", () => {
  it("should validate snapshot", () => {
    const { baseElement } = render(<Output commends={[]} />);
    expect(baseElement).toMatchSnapshot();
  });
  it("should validate snapshot with commends", () => {
    const { baseElement } = render(
      <Output commends={["place 1,2,west", "move"]} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it("should validate text", () => {
    render(<Output commends={["place 1,2,west", "move"]} />);

    const element = screen.getByText(/place 1,2,west/i);
    expect(element).toBeDefined();
  });
});
