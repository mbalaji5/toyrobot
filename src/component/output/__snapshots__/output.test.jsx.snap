// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should validate Error component should validate snapshot 1`] = `
<body>
  <div>
    <div
      class="output"
    />
  </div>
</body>
`;

exports[`Should validate Error component should validate snapshot 2`] = `
<body>
  <div>
    <div
      class="output"
    >
      <div>
        <span>
          &gt;
          place 1,2,west
        </span>
        <br />
      </div>
      <div>
        <span>
          &gt;
          move
        </span>
        <br />
      </div>
    </div>
  </div>
</body>
`;

exports[`Should validate Error component should validate snapshot with commends 1`] = `
<body>
  <div>
    <div
      class="output"
    >
      <div>
        <span>
          &gt;
          place 1,2,west
        </span>
      </div>
      <div>
        <span>
          &gt;
          move
        </span>
      </div>
    </div>
  </div>
</body>
`;
